cobrand_hackney/static/js.js

```diff
@@ -37,3 +37,69 @@ document.querySelectorAll('.case-detail__map__overlay').forEach(function(b) {
 });
 
 })();
+
+
+/* Creating a new filter called "Case Locations"  */ 
+// Fixes issue https://github.com/mysociety/noiseworks/issues/53
+
+// Creating a new dropdown "Case Locations"
+var caseLocationDiv = document.createElement('div');
+caseLocationDiv.className = 'govuk-form-group lbh-form-group';
+caseLocationDiv.id = 'div_id_case_location';
+caseLocationDiv.innerHTML = '<label for="id_case_location" class="govuk-label lbh-label">Case location</label>' + '<select class="govuk-select lbh-select" id="id_case_location"> <option value="all_areas" selected>All areas</option> <option value="selected_areas">Selected areas</option> <option value="outside_hackney">Outside Hackney</option> </select> ';
+
+// Defining the div that contains all the ward checkboxes
+var area = document.getElementById("div_id_ward");
+
+// Locating "Case Locations" before the area
+parentFormNode = area.parentNode
+parentFormNode.insertBefore(caseLocationDiv, area)
+
+// Defining the rest of the variables
+var inputArea = document.getElementById("id_case_location");
+var outsideHackney = document.getElementById("id_ward_22");
+var myAreasCheckboxChecked = document.querySelectorAll('.govuk-checkboxes__input:checked');
+var myAreasCheckbox = document.querySelectorAll('.govuk-checkboxes__input');
+var inputs = document.querySelectorAll('.govuk-checkboxes__input[name=ward]');
+area.style.display = "none";
+
+// Whenever there is at least one checkbox checked the filter will select by default "selected areas"
+// and will display all the checkboxes.
+if ( ((myAreasCheckboxChecked.length > 0 && myAreasCheckboxChecked.length < myAreasCheckbox.length ) || (myAreasCheckboxChecked.length > 1 && outsideHackney.checked == true)) && (myAreasCheckboxChecked.length != myAreasCheckbox.length) && !(myAreasCheckboxChecked.length == 1 && outsideHackney.checked == true) )  {
```

Comment:
I think (but am not sure, correct me if I’m wrong!) you could avoid this horribly complex condition here, by swapping the order of your `if` conditions, and leaving the `selected_areas` case till last, like so:

```javascript
if (outsideHackney.checked == true && myAreasCheckboxChecked.length == 1) {
    inputArea.value = 'outside_hackney';
} else if (myAreasCheckboxChecked.length == myAreasCheckbox.length) {
    inputArea.value = 'all_areas';
} else {
    inputArea.value = 'selected_areas';
    area.style.display = "block";
}
```

…I think?

Reply:
I'll check it tomorrow. It's definitely more efficient, so happy to use it if we can.

Reply:
I tried It and it worked great. I just made the changes and amended the pull request. 
Thanks for the feedback, Zarino.